types/block.go

```diff
@@ -1057,7 +1057,11 @@ func (data *Data) Hash() tmbytes.HexBytes {
 
 	// compute the data availability header
 	// todo(evan): add the non redundant shares back into the header
-	shares, _ := data.ComputeShares()
+	shares, _, err := data.ComputeShares(data.OriginalSquareSize)
+	if err != nil {
+		// todo(evan): see if we can get rid of this panic
+		panic(err)
```

Comment:
I think the only way this can panic is via a bug in nmt. And then it probably should. But we should definitely confirm this.

Reply:
left an issue for this here https://github.com/celestiaorg/celestia-core/issues/728

It might be reachable if a malicious node sends bad blocks while syncing, but am not sure. Even if it isn't reachable in normal use, we should still patch it so that we don't leave foot-guns lying around. This change will be easy, but quite large, so leaving it for another PR if that's ok